### ISSUE TITLE
New "prior" secondary calculation

### ISSUE THREAD

**Issue by Aylr** [dbt-labs/dbt_metrics#38]
To ease period over period reporting in some BI tools, an additional secondary calculation could be added.

Example usage:
```
{{ metrics.metric(
    metric_name='sessions',
    grain='day',
    dimensions=[],
    secondary_calculations=[
        metrics.period_over_period(comparison_strategy="prior", interval=7, alias="sessions_7_days_ago"),
    ],
) }}
```

This would perform no actual calculation but pull the metric value from the correct interval.

**Comment by Aylr** [comment]
I'm opening this issue to work on it as I have a proof of concept. I'm totally open to feedback on the idea or implementation.

**Comment by callum-mcdata** [comment]
Hi @Aylr! First of all, love the willingness to open an issue on something you see this package missing! I'd love to hear more about shape of the data that you're working with and how having a `sessions_7_days_ago` field is helpful to your BI tool. What BI tool are you working with?

As it stands for implementation, I think having a `prior` option for the comparison strategy 100% is the best implementation here - it fits neatly within `period_over_period`, can probably re-use a good chunk of the interval logic, and seems like an ergonomic way to write out the information.

Do you have a current working proof of concept?

**Comment by Aylr** [comment]
I do have a working proof of concept. I'm going to have a go at some integration tests to prove it works.

**Comment by callum-mcdata** [comment]
Sweet! Let me know if you run into any issues and I'm happy to help point you in the right direction. Also a heads up we're working on a pretty significant overhaul of the package (v0.3.0) in #37 so if you get this over the line before we merge that I'd just convert your integration tests into the new datasets there.

**Comment by callum-mcdata** [comment]
@Aylr closing this issue as we've merged it into the upcoming release in PR #127 !

**Comment by Aylr** [comment]
How exciting!
